samples/create_cluster.js

```diff
@@ -0,0 +1,164 @@
+ * backing off (delay) time used when retrying.
+ *
+ * @param {number} delay the previous delay in the sequence
+ * @returns the next delay before retrying
+ */
+function getFibonacciDelay(delay) {
+  const newDelay = prevFibonacciDelay + delay;
+  prevFibonacciDelay = delay;
+  return newDelay;
+}
+
+/**
+ * This method calls the Promise function (`checkOpStatus()`) to check the
+ * status of an operation and schedules a retry if the operation is still not
+ * completed. The retry delay is modelled as a fibonacci sequence.
+ *
+ * @param {container.v1.ClusterManagerClient} client the google cloud API client used to submit the request
+ * @param {string} opId the unique identifier of the operation we want to check
+ * @param {number} delay the delay with which the next retry is to be scheduled
+ * @param {number} maxtries maximum number of retry attempts to go before giving up
+ * @param {number} retryCount the current number of retried attempted
+ */
+function checkStatusWithRetry(client, opId, delay, maxtries, retryCount = 1) {
+  checkOpStatus(client, opId)
+    .then(status => {
+      // success scenario
+      console.log(status);
+    })
+    .catch(status => {
+      // fail (not yet complete) scenario
+      if (retryCount < maxtries) {
+        console.log(`${status} will try after ${delay / 1000}s delay...`);
+        const newDelay = getFibonacciDelay(delay);
+        setTimeout(
+          () =>
+            checkStatusWithRetry(
+              client,
+              opId,
+              newDelay,
+              maxtries,
+              retryCount + 1
+            ),
+          delay
+        );
+      } else {
+        console.log(`${status} max retries reached, giving up.`);
+      }
+    });
+}
+
+/**
+ * This method creates a GKE cluster using the given client in the GCP project
+ * and zone indicated by the `gcpProject` and `gcpZone` parameters. The last
+ * argment is used as the name given to the newly created cluster.
+ *
+ * @param {container.v1.ClusterManagerClient} client the google cloud API client used to submit the request
+ * @param {string} gcpProject the GCP project where the cluster is to be created
+ * @param {string} gcpZone the GCP zone where the cluster is to be created
+ * @param {string} clusterName the name to be given to the new cluster
+ */
+async function createCluster(gcpZone, clusterName) {
+  // Create the Cluster Manager Client
+  const client = new container.v1.ClusterManagerClient();
+  const gcpProject = await client.getProjectId();
+  const clusterLocation = `projects/${gcpProject}/locations/${gcpZone}`;
+  const request = {
+    parent: clusterLocation,
+    cluster: {
+      name: clusterName,
+      initialNodeCount: 2,
+      nodeConfig: {
+        machineType: 'e2-standard-2',
+      },
+    },
+  };
+  // invoke the create cluster API using the client
+  const [createOperation] = await client.createCluster(request);
```

Comment:
I could just utilized gax LRO feature, see [HOWTO](https://github.com/googleapis/gax-nodejs/blob/be8175624624648a32d535c3f6495814c163eb18/client-libraries.md#long-running-operations)
`const [resp] = await createOperation.promise()` to wait for the opertaion finished and get the `cluster`

Reply:
Part of the issue why these samples were required is because the container API protos have not been annotated correctly to ensure the default `Operation` objects are wrapped with `LRO Operation` that has the nice methods like `promise()`. See [b/189985582](http://b/189985582) and description of PR for more details.

So if I am not mistaken the return object of `client.createCluster(request);` is not the same `Operation` as [documented here](https://googleapis.dev/nodejs/google-gax/latest/classes/Operation.html). Hence, the reason we needed to provide sample to show how to handle this other Operation type.

Please correct me if I had understood the bug wrong!

Reply:
Wow. ok, then it make sense to me now. Thanks for explaining.